lib/evm/precompiles/types.ts

```diff
@@ -0,0 +1,29 @@
+import BN = require('bn.js')
+import Common from 'ethereumjs-common'
+const { ERROR } = require('../../exceptions.js')
+
+export interface PrecompileFunc {
+  (opts: PrecompileInput): PrecompileResult
+}
+
+export interface PrecompileInput {
+  data: Buffer;
+  gasLimit: BN;
+  _common: Common;
+}
+
+export interface PrecompileResult {
+  gasUsed: BN;
+  return: Buffer;
+  exception: 0 | 1;
+  exceptionError?: string;
+}
+
+export function OOGResult(gasLimit: BN): PrecompileResult {
+  return {
+    return: Buffer.alloc(0),
+    gasUsed: gasLimit,
+    exception: 0, // 0 means VM fail (in this case because of OOG)
+    exceptionError: ERROR.OUT_OF_GAS
+  }
+}
```

Comment:
Really really nice to have these new precompile interfaces and to have this more systematically structured and wrapped like this.